crates/solvers/src/tests/dex.rs

```diff
@@ -0,0 +1,113 @@
+use std::{
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+};
+
+#[derive(Debug, Clone)]
+pub enum Expectation {
+    Get {
+        path: String,
+        res: serde_json::Value,
+    },
+    Post {
+        path: String,
+        req: serde_json::Value,
+        res: serde_json::Value,
+    },
+}
+
+/// Set up an mock external DEX or DEX aggregator API.
+pub async fn setup(expectations: Vec<Expectation>) -> SocketAddr {
```

Comment:
Do you think we should rather only have 1 mock server implementation and also use this one for the `legacy` tests?

Reply:
Yes, I thought about this but didn't want change too much here. Will do it in a follow up.

Reply:
https://github.com/cowprotocol/services/pull/1244